blog/2022/05-10-resoto-graph-traversal/index.md

```diff
@@ -0,0 +1,119 @@
+
+```bash
+> search is(digitalocean_droplet) and id=289061882 --> is(volume)
+// highlight-start
+​kind=digitalocean_volume, name=pv2, age=2mo7d, account=10225075, region=Amsterdam 3
+​kind=digitalocean_volume, name=pv1, age=2mo7d, account=10225075, region=Amsterdam 3
+// highlight-end
+```
+
+This search finds the DigitalOcean droplet with the related id, then walks the graph outbound by one step, as defined by `-->` , and filters the resulting volume list. Other resources are pointing towards our droplet. By looking at the graph structure, we see it could be load balancers, kubernetes clusters, firewalls, or images. To get all the resources that point to this resource, we need to walk the graph inbound - which means in the opposite direction of the arrow in the graph structure diagram. Resoto uses the left facing arrow `<--` to walk the graph inbound. We use the same filter as before but this time, we walk the graph inbound and count the number of resources by kind.
+
+```bash
+> search is(digitalocean_droplet) and id=289061882 <-- | count kind
+// highlight-start
+​digitalocean_region: 1
+​digitalocean_image: 1
+​digitalocean_vpc: 1
+​digitalocean_kubernetes_cluster: 1
+​digitalocean_load_balancer: 1
+​total matched: 5
+​total unmatched: 0
+// highlight-end
+```
+
+The droplet belongs to a region inside a [VPC](https://docs.digitalocean.com/products/networking/vpc/). It is the target of a load balancer and part of a kubernetes cluster. Let's find out which other instances exist in the same VPC:
+
+```bash
+> search is(digitalocean_droplet) and id=289061882
+  <-- is(digitalocean_vpc) --> is(digitalocean_droplet)
+// highlight-start
+​id=289061882, name=dev-1, age=2mo7d, account=10225075, region=Amsterdam 3
+​id=289061881, name=dev-3, age=2mo7d, account=10225075, region=Amsterdam 3
+​id=289061880, name=dev-2, age=2mo7d, account=10225075, region=Amsterdam 3
+// highlight-end
+```
+
+We again start at our droplet and walk the graph inbound to the VPC. From there, we walk one step outbound to find all droplets attached to the same VPC. There are three droplets in this VPC, and one of them is the one we were starting from. It is possible to chain any number of graph traversals. We could have used the same technique to find all the droplets in the Kubernetes cluster or the ones behind the load balancer.
```

Comment:
I think you want to add 3 use cases here where the graph model comes in helpful, and where a simple flat inventory model falls short.

Reply:
I already show more than 3 use cases. What is missing?